challenge_8/C/randomDeepCopy.c

```diff
@@ -0,0 +1,111 @@
+/*
+Author: karanchawla
+Random Deep Copy
+#Challenge #8
+*/
+
+#include <stdio.h> 
+#include <stdlib.h> 
+
+
+//node definition for the linked list
+struct node
+{
+	int data;
+	struct node* next;
+	struct node* random;
+};
```

Comment:
This is *completely irrelevant* to the rest of the review, just a comment.

Best practice is to order structs from widest to narrowest. Since pointers are always a machine word across, they are usually the widest and first (there are types wider than a word, but except on 32-bit CPUs those aren't common). I, personally, would order this next, random, data.

It's not really important, and the compiler will make it all work out, but ordering widest to narrowest will have the fewest paddings inserted. The compiler may or may not reorder to fill space if you have many partial-word items, which you don't here, but that's why it's a good practice.